src/bn256/curve.rs

```diff
@@ -199,116 +199,75 @@ impl G1 {
 }
 
 #[cfg(test)]
-mod tests {
-    use crate::arithmetic::CurveEndo;
-    use crate::bn256::{Fr, G1, G2};
-    use crate::CurveExt;
-    use ff::Field;
-    use ff::{PrimeField, WithSmallOrderMulGroup};
-    use rand_core::OsRng;
+crate::curve_testing_suite!(G1, G2);
 
-    #[test]
-    fn test_hash_to_curve() {
-        crate::tests::curve::hash_to_curve_test::<G1>();
-    }
+#[cfg(test)]
+crate::curve_testing_suite!(G1, "hash_to_curve");
+
+#[cfg(test)]
+crate::curve_testing_suite!(G1, "endo_consistency");
+
+#[cfg(test)]
+crate::curve_testing_suite!(
+    G1,
+    "endo",
+    [
+        0x8b17ea66b99c90dd,
+        0x5bfc41088d8daaa7,
+        0xb3c4d79d41a91758,
+        0x00,
+    ]
```

Comment:
Can we import this from the EndoParams directly? 
I see it risky to duplicate constants arround.
Also, please add some comments if there's no alternative to duplication in order to know what that is.

Reply:
Um, @CPerezz 
Can you help me here?
I wanted to add the comments related to `z_other`.
But, I failed to find the parameter or algorithm related to `ZETA` and `z_other`. :cry: 
(Just updated the "endo" testing macro to receive the `z_other` param as optional [here](https://github.com/privacy-scaling-explorations/halo2curves/pull/129/commits/f7f5aa811dc5eb16547e9691c6bee5b8670960cd))

Reply:
Can't we compute it by ourseleves here or something similar? I think it makes it complex if we indeed need to pass more parameters to the macro.

Maybe if it gets too complex we can leave it as is and adding a comment saying how the number was obtained.

Reply:
@CPerezz @davidnevadoc 
Here, I tried to add the comments. https://github.com/privacy-scaling-explorations/halo2curves/pull/129/commits/9ee6301198c99842826277ae13bebb5db905db03